### PR TITLE
umhs-28: use canonical URL for results

### PR DESCRIPTION
**Pull request by jesconstantine**

### CODE REVIEW

**Comment by agentrickard** [comment]
I think this should work. Requires a patch to Domain in D7.

See https://www.drupal.org/project/domain/issues/3047545

The new plugin should return a URL or NULL if the URL should be relative to the requesting site.

**Comment by agentrickard** [comment]
I'm going to rewire this to not require a new Domain Access release.

File: src/SearchApiFederatedSolrCanonicalUrl.php [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+      // Get the entity object for the item being indexed, exit if there's somehow not one.
+      $entity = current(entity_load($entity_type, [$id]));
+      if (!$entity) {
+        return;
+      }
+
+      // Determine if there is a canonical URL for the content.
+      // This only comes into play if domain source is used.
+      if (isset($node->domain_source) && $node->domain_source == DOMAIN_SOURCE_USE_ACTIVE) {
+        $url = NULL;
```
Comment (agentrickard) [inline src/SearchApiFederatedSolrCanonicalUrl.php]: Set to `''`

**Comment by jesconstantine** [comment]
Tested by Ken locally and by Paul from Umich on their dev env

**Pull request merged**